### ISSUE TITLE
RCS1202 Avoid NullRef - Needs to be more strict

### ISSUE THREAD

**Issue by zitcomdev** [JosefPihrt/Roslynator#704]
**Steps to Reproduce**:
```csharp
interface Foo {
  bool? Get();
}

class Bar {
  void DoStuff(Foo foo) {
    var x = foo.Get().Value;
  }
}
```

**Actual Behavior**:
RCS1202 Avoid NullRef not triggered

**Expected Behavior**:
RCS1202 Avoid NullRef should be triggered

**Other**:
Think RCS1202 should be a warning by default.
